addons/html_editor/static/tests/sanitize.test.js

```diff
@@ -35,3 +39,80 @@ test("sanitize plugin should handle aria-label attribute with data-oe-aria-label
         contentAfter: `<p data-oe-aria-label="status">a[]</p>`,
     });
 });
+
+test("fixInvalidHTML should close self-closing elements", () => {
+    expect(fixInvalidHTML(markup`<t/>`).toString()).toBe("<t></t>");
+    expect(fixInvalidHTML(markup`<t class="test"/>`).toString()).toBe('<t class="test"></t>');
+    expect(fixInvalidHTML(markup`<a/>`).toString()).toBe("<a></a>");
+    expect(fixInvalidHTML(markup`<a href="#"/>`).toString()).toBe('<a href="#"></a>');
+    expect(fixInvalidHTML(markup`<strong/>`).toString()).toBe("<strong></strong>");
+    expect(fixInvalidHTML(markup`<strong class="bold"/>`).toString()).toBe(
+        '<strong class="bold"></strong>'
+    );
+    expect(fixInvalidHTML(markup`<span/>`).toString()).toBe("<span></span>");
+    expect(fixInvalidHTML(markup`<span id="test"/>`).toString()).toBe('<span id="test"></span>');
+    expect(
+        fixInvalidHTML(
+            markup`<t t-out="object.name"/>asdf<t t-out="object.parner_id.name"/>`
+        ).toString()
+    ).toBe('<t t-out="object.name"></t>asdf<t t-out="object.parner_id.name"></t>');
+});
+
+test("fixInvalidHTML escapes string input", () => {
+    expect(fixInvalidHTML("<t/>").toString()).toBe("&lt;t/&gt;");
+    expect(fixInvalidHTML('<a href="?param=value&other=test"/>').toString()).toBe(
+        "&lt;a href=&quot;?param=value&amp;other=test&quot;/&gt;"
+    );
```

Comment:
The tags are still self-closing here. This is a trap waiting to be sprung on any unsuspecting developer because they might end up calling `fixInvalidHtml` with string input instead of markup and they will not realize that it does _not_ fix the html in that case.

Reply:
The HTML is actually fixed in that case. Without markup it's just assimilate to a simple text node with text content only, which doesn't need any fixing.

Reply:
I see your point on a theoretical basis, but in practice I still think it's a trap. That being said, the developer will see it since the html ends up escaped so it won't work when they try to use it as actual html. It's ok for me then.